internal/trie/node/leaf.go

```diff
@@ -14,13 +14,14 @@ var _ Node = (*Leaf)(nil)
 
 // Leaf is a leaf in the trie.
 type Leaf struct {
-	Key   []byte // partial key
+	// Partial key bytes in nibbles (0 to f in hexadecimal)
+	Key   []byte
 	Value []byte
 	// Dirty is true when the branch differs
 	// from the node stored in the database.
-	dirty      bool
+	Dirty      bool
 	hashDigest []byte
-	encoding   []byte
+	Encoding   []byte
```

Comment:
why are these exported now? i couldn't really see why. same for branch

Reply:
For testing in the `lib/trie` package, to be able to assert/expect leaves/branches and set their encoding without weird wrappers. I have it locally but wanted to have this PR first to split the work since so the next test PR will be just test code.